diff --git a/gl/lib/getprogname.c b/gl/lib/getprogname.c
index 62a4800..b3986db 100644
--- a/gl/lib/getprogname.c
+++ b/gl/lib/getprogname.c
@@ -213,8 +213,12 @@ getprogname (void)
               if (token > 0 && buf.ps_pid == pid)
                 {
                   char *s = strdup (last_component (buf.ps_pathptr));
-                  if (s)
+                  if (s) {
                     p = s;
+#if (__CHARSET_LIB == 1)
+                    __e2a_s(p);
+#endif
+                  }
                   break;
                 }
             }
