diff --git a/src/man.c b/src/man.c
index 761f4b4..88ff6f1 100644
--- a/src/man.c
+++ b/src/man.c
@@ -342,12 +342,14 @@ static struct argp_option options[] = {
 	{ 0 }
 };
 
+# ifdef TROFF_IS_GROFF
 static void init_html_pager (void)
 {
 	html_pager = getenv ("BROWSER");
 	if (!html_pager)
 		html_pager = PROG_BROWSER;
 }
+#endif
 
 static error_t parse_opt (int key, char *arg, struct argp_state *state)
 {
