diff --git a/configure b/configure
index 108ad4c..d321125 100755
--- a/configure
+++ b/configure
@@ -11743,7 +11743,9 @@ case $host_os in
 aix[4-9]*)
   lt_cv_deplibs_check_method=pass_all
   ;;
-
+openedition)
+  lt_cv_deplibs_check_method=pass_all
+  ;;
 beos*)
   lt_cv_deplibs_check_method=pass_all
   ;;
@@ -18950,7 +18952,10 @@ EOF
 \\n(.X
 EOF
 `
-   if test "x$man_heirloom_ext_plain" != 0 && \
+   # Basic return code test
+   $nroff -mg
+
+   if test $? -eq 0 && test "x$man_heirloom_ext_plain" != 0 && \
       test "x$man_heirloom_ext_mg" != 0 && \
       test "x$man_heirloom_ext_plain" != "x$man_heirloom_extflag_mg"
    then
