diff --git a/gl/lib/getprogname.h b/gl/lib/getprogname.h
index 13314f2..db76ca6 100644
--- a/gl/lib/getprogname.h
+++ b/gl/lib/getprogname.h
@@ -19,8 +19,13 @@
 
 #include <stdlib.h>
 
+
 #if __GNUC__ || (__clang_major__ >= 4)
 # warning "The include file getprogname.h is deprecated. Use <stdlib.h> instead."
+#else
+# ifdef __MVS__
+   extern char const *getprogname (void);
+# endif
 #endif
 
 #endif
