diff --git a/src/man.c b/src/man.c
index 761f4b4..2465861 100644
--- a/src/man.c
+++ b/src/man.c
@@ -342,12 +342,14 @@ static struct argp_option options[] = {
 	{ 0 }
 };
 
+# ifdef TROFF_IS_GROFF
 static void init_html_pager (void)
 {
 	html_pager = getenv ("BROWSER");
 	if (!html_pager)
 		html_pager = PROG_BROWSER;
 }
+#endif
 
 static error_t parse_opt (int key, char *arg, struct argp_state *state)
 {
@@ -4417,8 +4419,25 @@ int main (int argc, char *argv[])
 					if (!section && maybe_section &&
 					    CTYPE (isdigit, nextarg[0]))
 						gripe_no_name (nextarg);
-					else
-						gripe_no_man (nextarg, section);
+					else {
+#ifdef __MVS__
+            char* fmt = "/bin/man %s %s 2>/dev/null";
+            char* cmd = malloc(strlen(nextarg)+strlen(section)+strlen(fmt)+1);
+            int rc = 0;
+            if (cmd) {
+              sprintf(cmd, fmt, section, nextarg);
+              rc = system(cmd);
+            }
+            if (!cmd || rc) {
+					    gripe_no_man (nextarg, section);
+            }
+            if (cmd) {
+              free(cmd);
+            }
+          }
+#else
+					gripe_no_man (nextarg, section);
+#endif
 				}
 			}
 		} else {
