diff --git a/src/man.c b/src/man.c
index 5d3be39..6d71087 100644
--- a/src/man.c
+++ b/src/man.c
@@ -4479,8 +4479,25 @@ int main (int argc, char *argv[])
 					if (!section && maybe_section &&
 					    CTYPE (isdigit, nextarg[0]))
 						gripe_no_name (nextarg);
-					else
-						gripe_no_man (nextarg, section);
+					else {
+#if __MVS__
+            char* fmt = "/bin/man %s %s 2>/dev/null";
+            char* cmd = malloc(strlen(nextarg)+strlen(section)+strlen(fmt)+1);
+            int rc = 0;
+            if (cmd) {
+              sprintf(cmd, fmt, section, nextarg);
+              rc = system(cmd);
+            }
+            if (!cmd || rc) {
+					    gripe_no_man (nextarg, section);
+            }
+            if (cmd) {
+              free(cmd);
+            }
+          }
+#else
+					gripe_no_man (nextarg, section);
+#endif
 				}
 			}
 		} else {
